src/main/kotlin/exercise2/task3/ConvertMethodToFunctionalApproach.kt

```diff
@@ -17,24 +17,29 @@ import kotlin.math.abs
  */
 
 internal fun List<Int>.findPairWithBiggestDifference(): Pair<Int, Int> {
-    // TODO refactor me to functional approach and make tests pass!!!
-    var resultPair: Pair<Int, Int>? = null
-    var biggestDifference = Int.MIN_VALUE
-
-    for (i in this.indices) {
-        for (j in (i + 1) until this.size) {
-            val first = this[i]
-            val second = this[j]
-            val absDifference = abs(first - second)
-
-            if (absDifference >= biggestDifference) {
-                biggestDifference = absDifference
-                resultPair = Pair(first, second)
-            }
-        }
-    }
 
-    return resultPair!!
+    return Pair(
+        this.sorted().get(this.size-1), this.sorted().get(0)
```

Comment:
The solution is correct, but it is written in Java-style code. Please consider my suggestion above to implement it in a Kotlin, functional style. :) 

Also, `this.size - 1` has a Kotlin equivalent in `this.lastIndex`, and `get(0)` has an equivalent in a `first()` function.